src/main.cpp

```diff
@@ -463,10 +464,16 @@ static wf_recorder_output* choose_interactive()
     fflush(stdout);
 
     int choice;
-    if (scanf("%d", &choice) != 1 || choice > (int)available_outputs.size() || choice <= 0)
-        return nullptr;
+    scanf("%d", &choice);
 
-    return &available_outputs[choice - 1];
+    i = 1;
```

Comment:
I'd rather use `std::advance` here, gives you better readability. Also, you'd need to keep the check above.

Reply:
Ah yeah this is a lot nicer :)